src/util.js

```diff
@@ -588,3 +588,120 @@ export function dateIsEmpty(date) {
+ * @param month A zero-based month.
+ * @param day A one-based day.
+ * @returns A date object.
+ */
+function dateFromUTC(year, month, day) {
+  const date = new Date(Date.UTC(year, month, day))
+  // Date.UTC() treats a two-digit year as an offset from 1900.
+  date.setUTCFullYear(year)
+  return date
+}
+
+/**
+ * Converts the given ISO 8601-1 date to a decimal year.
+ *
+ * @param isoDate A date string in ISO 8601-1 format.
+ * @returns A floating point number of years since year 0.
+ */
+function decimalYearFromISODate(isoDate) {
+  // Require a valid YYYY, YYYY-MM, or YYYY-MM-DD date, but allow the year
+  // to be a variable number of digits or negative, unlike ISO 8601-1.
+  if (!isoDate || !/^-?\d{1,4}(?:-\d\d){0,2}$/.test(isoDate)) return undefined
+
+  const ymd = isoDate.split('-')
+  // A negative year results in an extra element at the beginning.
+  if (ymd[0] === '') {
+    ymd.shift()
+    ymd[0] *= -1
+  }
+  const year = +ymd[0]
+  const date = dateFromUTC(year, +ymd[1] - 1, +ymd[2])
+  if (Number.isNaN(date)) return undefined
+
+  // Add the year and the fraction of the date between two New Year’s Days.
+  const nextNewYear = dateFromUTC(year + 1, 0, 1).getTime()
+  const lastNewYear = dateFromUTC(year, 0, 1).getTime()
+  return year + (date.getTime() - lastNewYear) / (nextNewYear - lastNewYear)
+}
+
+/**
+ * Returns a modified version of the given filter that only evaluates to
+ * true if the feature coincides with the given decimal year.
+ *
+ * @param filter The original layer filter.
+ * @param decimalYear The decimal year to filter by.
+ * @returns A filter similar to the given filter, but with added conditions
+ *	that require the feature to coincide with the decimal year.
+ */
+function constrainFilterByDate(filter, decimalYear) {
+  const newFilter = filter
+  if (filter && filter[0] === 'all' && filter[1] && filter[1][0] === 'any') {
+    if (
+      filter[1][2] &&
+      filter[1][2][0] === '<=' &&
+      filter[1][2][1] === 'start_decdate'
+    ) {
+      newFilter[1][2][2] = decimalYear
+    }
+    if (
+      newFilter[2][2] &&
+      newFilter[2][2][0] === '>=' &&
+      newFilter[2][2][1] === 'end_decdate'
+    ) {
+      newFilter[2][2][2] = decimalYear
+    }
+    return newFilter
+  }
+
+  const dateFilter = [
+    'all',
+    ['any', ['!has', 'start_decdate'], ['<=', 'start_decdate', decimalYear]],
+    ['any', ['!has', 'end_decdate'], ['>=', 'end_decdate', decimalYear]],
+  ]
+  if (filter) {
+    dateFilter.push(filter)
+  }
+  return dateFilter
+}
+
+/**
+ * Filters the map’s features by the `date` data attribute.
+ *
+ * @param map The MapboxGL map object to filter the style of.
+ * @param year The numeric ear to filter by
+ */
+export function filterByDecimalYear(map, decimalYear) {
+  // eslint-disable-next-line array-callback-return
+  map.getStyle().layers.map(layer => {
+    if (!('source-layer' in layer)) return
+
+    const filter = constrainFilterByDate(layer.filter, decimalYear)
```

Comment:
> I changed it to a normal slider now and added the filter methods; however it doesn't seem to work yet; or it works only once, to be precise. Not sure what I'm doing wrong.

Ah, the original embed code was only designed to filter the map initially without any updates. OpenHistoricalMap/ohm-embed-1yr#5 will fix this issue in order to implement animation in that embed. The relevant part of that PR is changing this line to get the most up-to-date filter, rather than the one that happens to have been in the layer the last time we got the style:

```suggestion
    const filter = constrainFilterByDate(map.getFilter(layer.id), decimalYear)
```